setup.py

```diff
@@ -53,6 +55,21 @@ def get_source_files(self):
         return filenames
 
     def build_extension(self, ext):
+        if ext.sources is None or not isinstance(ext.sources, (list, tuple)):
+            from distutils.errors import DistutilsSetupError
+            raise DistutilsSetupError(
+                "in 'ext_modules' option (extension '%s'), "
+                "'sources' must be present and must be "
+                "a list of source filenames" % ext.name)
+
+        ext_path = self.get_ext_fullpath(ext.name)
+        depends = ext.sources + ext.depends
+        if not (self.force or newer_group(depends, ext_path, 'newer')):
```

Comment:
don't know.. I just copied from the default distutils `build_ext` module. I thought it'd be better to keep the default behavior. But I see that with "error" we could have avoided issues like the missing comma..